### PR TITLE
use packaged asset for GnosisSafeProxyFactory

### PR DESCRIPTION
**Pull request by fleupold**
As #528 but this time for GnosisSafeProxyFactory

### Test Plan

CI

### CODE REVIEW

**Review by bh2smith — APPROVED** [review]
Sweet sauce! Are there any others like this coming soon?

**Comment by fleupold** [comment]
> Are there any others like this coming soon?

Yes, MultiSend and BatchExchange dependency are still missing.

**Comment by bh2smith** [comment]
> Yes, MultiSend and BatchExchange dependency are still missing.

I would be happy to take on one of these today if you don't mind.

File: migrations/2_dev_dependencies.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const Artifactor = require("@truffle/artifactor")
 const migrateBatchExchange = require("@gnosis.pm/dex-contracts/src/migration/PoC_dfusion")
 const { GnosisSafe } = require("../scripts/utils/dependencies")(web3, artifacts)
-const ProxyFactory = artifacts.require("./GnosisSafeProxyFactory.sol")
+const { GnosisSafeProxyFactory } = require("../scripts/utils/dependencies")(web3, artifacts)
```
Comment (bh2smith) [inline migrations/2_dev_dependencies.js:4]: Can these both be imported in the same line?

Comment (fleupold) [reply migrations/2_dev_dependencies.js:4]: Yes they can...

File: migrations/2_dev_dependencies.js [hunk 2]
```diff
@@ -28,7 +27,9 @@ module.exports = async function (deployer, network, accounts) {
 }
 
 async function deploySafe(deployer) {
```
Comment (bh2smith) [inline migrations/2_dev_dependencies.js:29]: If we plan to move other items into here like Multisend, perhaps this function could use a rename `deploySafeItems`. Or is Multisend something that would continue to live outside?

**Comment by fleupold** [comment]
> I would be happy to take on one of these today if you don't mind.

Sorry I just saw your comment after already having converted the MultiSend one. 
But please go ahead with the BatchExchange one. It should be very similar (just need to use another Artefactor with a path to the dex-contracts repo as the build info is not in the safe package).

File: migrations/2_dev_dependencies.js (hunk 2 above)
Comment (fleupold) [reply migrations/2_dev_dependencies.js:29]: Done in #537

**Review by fedgiac — APPROVED** [review]
Looks good!

**Pull request merged**